File: umuganda/src/main/java/com/webtech/umuganda/core/location/service/LocationService.java
package com.webtech.umuganda.core.location.service;

import com.webtech.umuganda.core.location.dto.LocationDto;
import com.webtech.umuganda.core.location.model.Locations;

import java.util.List;
import java.util.UUID;

public interface LocationService {

    LocationDto createLocation(LocationDto dto);

    LocationDto updateLocation(UUID id, LocationDto dto);

    void deleteLocation(UUID id);

    LocationDto getLocationById(UUID id);

    List<LocationDto> getAllLocations();

    List<LocationDto> getChildren(UUID parentId);

    List<LocationDto> getByType(String type);
}

File: umuganda-frontend/src/services/locationService.js
import api from './api';

const locationService = {
    getAllLocations: async () => {
        const response = await api.get('/locations');
        return response.data;
    },

    getLocationById: async (id) => {
        const response = await api.get(`/locations/${id}`);
        return response.data;
    },

    getChildren: async (parentId) => {
        const response = await api.get(`/locations/parent/${parentId}`);
        return response.data;
    },

    getByType: async (type) => {
        const response = await api.get(`/locations/type/${type}`);
        return response.data;
    },

    createLocation: async (locationData) => {
        const response = await api.post('/locations', locationData);
        return response.data;
    },

    updateLocation: async (id, locationData) => {
        const response = await api.put(`/locations/${id}`, locationData);
        return response.data;
    },

    deleteLocation: async (id) => {
        await api.delete(`/locations/${id}`);
    }
};

export default locationService;
